src/components/select/SelectCommons.tsx

```diff
@@ -46,7 +47,7 @@ export interface SelectBaseProps {
   labelNotFound?: MantineSelectProps['nothingFound'];
   limit?: MantineSelectProps['limit'];
   name?: MantineSelectProps['name'];
-  optionComponent?: MantineSelectProps['itemComponent'];
+  optionComponent?: (selectItemOptions: SelectOption) => React.JSX.Element | null;
```

Comment:
I think we need a generic here, otherwise the typing of the component in the products is gonna be hard when you want to add more stuff on your options.

Reply:
I think it's fine: SelectOption includes `[key: string]: any;` so it fits any type that has a `value: string`

Reply:
Ah yeah nice catch so I think we are good, it's simpler not to have to rely on generic